src/libs/vehicle/ardupilot/ardupilot.ts

```diff
@@ -125,26 +131,68 @@ export abstract class ArduPilotVehicle<Modes> extends Vehicle.AbstractVehicle<Mo
     param4?: number,
     param5?: number,
     param6?: number,
-    param7?: number
-  ): void {
-    const command: Message.CommandLong = {
-      type: MAVLinkType.COMMAND_LONG,
-      param1: param1 ?? 0,
-      param2: param2 ?? 0,
-      param3: param3 ?? 0,
-      param4: param4 ?? 0,
-      param5: param5 ?? 0,
-      param6: param6 ?? 0,
-      param7: param7 ?? 0,
-      command: {
-        type: mav_command,
-      },
-      target_system: this.currentSystemId,
-      target_component: 1,
-      confirmation: 0,
-    }
+    param7?: number,
+    waitForAck = true
+  ): Promise<CommandAck> {
+    return new Promise((resolve, reject) => {
+      // Construct the command
+      const command: Message.CommandLong = {
+        type: MAVLinkType.COMMAND_LONG,
+        param1: param1 ?? 0,
+        param2: param2 ?? 0,
+        param3: param3 ?? 0,
+        param4: param4 ?? 0,
+        param5: param5 ?? 0,
+        param6: param6 ?? 0,
+        param7: param7 ?? 0,
+        command: {
+          type: mav_command,
+        },
+        target_system: this.currentSystemId,
+        target_component: 1,
+        confirmation: 0,
+      }
+
+      this.write(command)
+      if (!waitForAck) {
+        resolve(new CommandAck())
+        return
+      }
+
+      let ackReceived = false
+      const ackHandler = (commandAck: CommandAck): void => {
+        console.log('Received command acknowledgment:', commandAck)
+        if (commandAck.command.type === mav_command) {
+          ackReceived = true
+          this.onCommandAck.remove(ackHandler)
 
-    this.write(command)
+          if (![MavResult.MAV_RESULT_ACCEPTED, MavResult.MAV_RESULT_IN_PROGRESS].includes(commandAck.result.type)) {
+            commandAck.alertMessage()
```

Comment:
Now that `alertMessage` has its own check for ack type, this should be done before the if line.

Reply:
I wanted to get rid of this check but, if I don't, a swal will be emitted on all command ACKs including the accepted commands. I thought of only having alerts for failed ACKs, but I thought it would be useful in certain cases to call the alert on an accepted ACK.